ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -45,7 +45,14 @@ cpdef traceback_to_frames(traceback, max_nframes):
             frame = tb.tb_frame
             code = frame.f_code
             lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.insert(0, DDFrame(code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
+
+            # If the class_name is nonempty, then prepend the classname to the function name like class.function
+            # We'll ignoer the old placeholder for the class_name for now
+            class_name = _extract_class_name(frame)
+            if class_name:
+                frames.append(DDFrame(code.co_filename, lineno, class_name + "." + code.co_name))
```

Comment:
Starting with CPython 3.11 we have `co_qualname` that we can use

Reply:
Good stuff, added support in ef884ae68